src/strategies/BeefyStrategy.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+// import "forge-std/console.sol";
+
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {ICurvePool, I3CrvMetaPoolZap} from "src/interfaces/curve.sol";
+import {IBeefyVault} from "src/interfaces/Beefy.sol";
+
+import {AccessStrategy} from "src/strategies/AccessStrategy.sol";
+
+import {Vault} from "src/vaults/Vault.sol";
+
+contract BeefyStrategy is AccessStrategy {
+    using FixedPointMathLib for uint256;
+    using SafeTransferLib for ERC20;
+
+    // curve pool info
+    int128 public immutable assetIndex;
+    ICurvePool public immutable curvePool;
+    I3CrvMetaPoolZap public immutable zapper;
+
+    // beefy vault
+    IBeefyVault public immutable beefy;
+    // const max bps
+    uint256 public constant MAX_BPS = 10_000;
+    uint256 public defaultSlippageBps;
+
+    constructor(
+        Vault _vault,
+        ICurvePool _pool,
+        I3CrvMetaPoolZap _zapper,
+        int128 _assetIndex,
+        IBeefyVault _beefy,
+        address[] memory strategists
+    ) AccessStrategy(_vault, strategists) {
+        assetIndex = _assetIndex;
+        curvePool = _pool;
+        zapper = _zapper;
+        beefy = _beefy;
+
+        require(address(beefy.want()) == address(curvePool), "BS: want asset mismatch");
+
+        curvePool.approve(address(beefy), type(uint256).max);
+        asset.approve(address(zapper), type(uint256).max);
+        curvePool.approve(address(zapper), type(uint256).max);
+    }
+
+    function _afterInvest(uint256 assets) internal override {
+        investIntoBeefy(assets, defaultSlippageBps);
+    }
+
+    function investIntoBeefy(uint256 assets, uint256 slippageBps) internal {
+        uint256[4] memory amounts = [uint256(0), 0, 0, 0];
+        amounts[uint256(uint128(assetIndex))] = assets;
+
+        // calc min lp token to receive with slippage
+        uint256 lpToken = zapper.calc_token_amount(address(curvePool), amounts, true);
+        uint256 minLpToken = lpToken.mulDivDown(MAX_BPS - slippageBps, MAX_BPS);
+
+        // add liquidity in curve pool
+        zapper.add_liquidity(address(curvePool), amounts, minLpToken);
+
+        // deposit lpToken in beefy
+        beefy.depositAll();
+    }
+
+    function _divest(uint256 assets) internal override returns (uint256) {
+        uint256 amount = divestFromBeefy(assets, defaultSlippageBps);
+        uint256 amountToSend = Math.min(assets, amount);
+        asset.safeTransfer(address(vault), amountToSend);
+        return amountToSend;
+    }
+
+    function divestFromBeefy(uint256 assets, uint256 slippageBps) internal returns (uint256) {
```

Comment:
This function probably doesn't need to return anything. After liquidation, the vault will have balance >= assets or balance < assets. The amountToSend logic makes sure that the correct number is returned.